### PR TITLE
Remove dependency on libbeat/cmd

### PR DESCRIPTION
**Pull request by axw**
## Motivation/summary

Remove our dependency on `libbeat/cmd` and `libbeat/cmd/instance`, inlining the parts of that code that we need. This is a precursor to addressing https://github.com/elastic/apm-server/issues/8928

This will give us more control over how APM Server is initialised and how it interacts with libbeat. For example:
 - we can look at moving our automaxprocs logic out of `beater` (which may be started and stopped) and into the `beatcmd.Beat.Run` method
 - we can eagerly fetch the Elasticsearch cluster UUID in the `beatcmd` code, allowing us to remove this concern from beater and undo https://github.com/elastic/apm-server/pull/6591
 - we can use `apm-server version` command can use `runtime/debug` for obtaining the commit and timestamp as described in https://github.com/elastic/apm-server/issues/8644 (this is done)

Another benefit is that the `apm-server` binary is smaller and quicker to build, due to not importing all the libbeat processors (which we do not support anyway) and their dependencies. On main apm-server-linux-amd64 is 70MB, and on this branch it is 45MB.

## Checklist

~- [ ] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/main/CHANGELOG.asciidoc)~
~- [ ] Update [package changelog.yml](https://github.com/elastic/apm-server/blob/main/apmpackage/apm/changelog.yml) (only if changes to `apmpackage` have been made)~
~- [ ] Documentation has been updated~

## How to test these changes

This is a non-functional change.

Check all `apm-server` subcommands continue to work as expected (keystore, apikey, version, etc.)

## Related issues

https://github.com/elastic/apm-server/issues/8928

### CODE REVIEW

File: internal/beatcmd/testcmd.go [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+	beaterconfig "github.com/elastic/apm-server/internal/beater/config"
+	"github.com/elastic/beats/v7/libbeat/outputs"
+	"github.com/elastic/elastic-agent-libs/config"
+	"github.com/elastic/elastic-agent-libs/testing"
+)
+
+func genTestCmd(beatParams BeatParams) *cobra.Command {
+	exportCmd := &cobra.Command{
+		Use:   "test",
```
Comment (lahsivjar) [inline internal/beatcmd/testcmd.go:33]: TIL

**Review by lahsivjar — APPROVED** [review]
LGTM! (I lack understanding of some of the `libbeat` components and what they do but overall looks good to me)

File: internal/beatcmd/keystore.go [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+var keystoreCommand = &cobra.Command{
+	Use:   "keystore",
+	Short: "Manage secrets keystore",
+}
+
+func init() {
+	keystoreCommand.AddCommand(genCreateKeystoreCmd())
+	keystoreCommand.AddCommand(genAddKeystoreCmd())
+	keystoreCommand.AddCommand(genRemoveKeystoreCmd())
+	keystoreCommand.AddCommand(genListKeystoreCmd())
```
[the remark this answers is not in the document]
Comment (axw) [reply internal/beatcmd/keystore.go:46]: Users can invoke `apm-server keystore` commands to manage secrets, and then reference them in `apm-server.yml`. See https://www.elastic.co/guide/en/apm/guide/current/keystore.html

**Pull request merged**